### ISSUE TITLE
No able to obtain proper result from "      Metrics::classification_report(y_train, train_predictions);"

### ISSUE THREAD

**Issue by Guitch82** [MegaJoctan/MALE5#7]
With the last versionof MALE5 3-0.0 I got issue to compile the example previously share. I ahve updated the code to finally manage to compile : 

/--- Normalizing the independent variables
    StandardizationScaler Scaler1;
   
    X_train = Scaler1.fit_transform(X_train); // we fit the scaler on the training data and transform the data alltogether
    X_test = Scaler1.transform(X_test); // we transform the new data this way
     
     Print("X_Train == ", X_train);

//--- Training the  model
     
     decision_tree.fit(X_train, y_train); //The training function 
     
//--- Measuring predictive accuracy 
   
     vector train_predictions = decision_tree.predict(X_train);
     
     Print("GC train prediction vector  == ", train_predictions);
     
     Print("Training results classification report");
     Metrics::classification_report(y_train, train_predictions);

//--- Evaluating the model on out-of-sample predictions
     
     vector test_predictions = decision_tree.predict(X_test);
     
     Print("Testing results classification report");
     Metrics::classification_report(y_test, test_predictions); 
     
//---

    ArraySetAsSeries(rates, true);

However it looks like this step doesn't work as expected      Metrics::classification_report(y_train, train_predictions). COuld you support?

**Comment by MegaJoctan** [comment]
can you show the current outcome you are receiving from the classification report, and what were you expecting to see

**Comment by Guitch82** [comment]
if fact I see in the tester that the confusion matrix is full of zero ...
![Capture d’écran du 2024-10-22 12-27-40](https://github.com/user-attachments/assets/1719b76e-f0e3-4976-894b-ef3c7f137d4f)

**Comment by Guitch82** [comment]
then I see the classification report full of zero as well..
![Capture d’écran du 2024-10-22 12-29-38](https://github.com/user-attachments/assets/f2b4282b-cbe0-4ad2-8958-dc10e72df114)

**Comment by Guitch82** [comment]
then I can see oall other valauble to zero. I enter in the function file Metrics to try to understant where wome from the issue... by adding some "print".
![Capture d’écran du 2024-10-22 12-31-11](https://github.com/user-attachments/assets/55401fbb-53c4-42c9-84a5-eb451256b790)
thanks for your support

**Comment by MegaJoctan** [comment]
It seems that you are using the classification report for a regression problem. 
That is where the issue lies.
Check your predictions to make sure they are binaries(classes). The classification report is for classification problems.

**Comment by Guitch82** [comment]
Thanks for your reply. I have investigated more in details. In fact  , I had to create a new function to be able to enter "y" vector that was a binary one. In MatrixExtend the option to select the "y" vector disappeared and per default "y" was the last column from the matrix X ..that didn't match my code. 

in your exemple you also used this function with "y" vector but it is not available anymore in your version MAT5 3.0.0
 MatrixExtend::TrainTestSplitMatricesGC(X, y,X_train, y_train, X_test, y_test, train_size, random_state); 

In anycase thanks for your answer and let see if I can manage to use this library efficiently ;-).
